lib/archive.go

```diff
@@ -227,21 +234,36 @@ func (arc *Archive) Write(out io.Writer) error {
 		// - We don't want to leak private information (eg. usernames) in archives, so make sure to
 		//   anonymize paths before stuffing them in a shareable archive.
 		foundDirs := make(map[string]bool)
-		paths := make([]string, 0, len(entry.files))
-		files := make(map[string][]byte, len(entry.files))
-		for filePath, data := range entry.files {
-			filePath = NormalizeAndAnonymizePath(filePath)
-			files[filePath] = data
-			paths = append(paths, filePath)
-			dir := path.Dir(filePath)
-			for {
-				foundDirs[dir] = true
-				idx := strings.LastIndexByte(dir, os.PathSeparator)
-				if idx == -1 {
-					break
+		paths := make([]string, 0, 10)
+		infos := make(map[string]os.FileInfo) // ... fix this ?
+		files := make(map[string][]byte)
+
+		err = fsext.Walk(filesystem, afero.FilePathSeparator,
+			filepath.WalkFunc(func(filePath string, info os.FileInfo, err error) error {
```

Comment:
I have no issues with this being a lambda, but can you assign it to a variable below `files`, so it's a bit more readable and one indentation level less?